tools/test/h5ls/expected/tfloat16.ls

```diff
@@ -0,0 +1,8 @@
+Opened "tfloat16.h5" with sec2 driver.
+DS16BITS                 Dataset {8/8, 16/16}
+    Attribute: DS16BITS {128}
+        Type:      native _Float16
```

Comment:
Will this be different on a big-endian machine?

Reply:
Ah that's a good point. I should test this on a big-endian machine, but I suspect that yes it will probably trip over this test.

Reply:
Added a BE test file for h5dump, h5dump xml and h5ls tests and verified testing on big-endian machine, but only in the case where `_Float16` isn't supported since we don't have a BE machine where it is. However, I setup the tests to only check the appropriate BE or LE test file, depending on the target architecture, for `native _Float16` and verified the correct file is picked between BE and LE machines with both CMake and Autotools, so this should work on BE machines where `_Float16` is supported.